From 5758cd040588d0fec656ee79d690920f894968ec Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Wed, 5 Oct 2016 12:17:54 -0400
Subject: [PATCH 086/134] trace-graph: Show events missed in cpu plot

If events were dropped, when hovering over the CPU plot, show that events
were missed.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-plot-cpu.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/trace-plot-cpu.c b/trace-plot-cpu.c
index eab5656c2a72..d83b595b78ff 100644
--- a/trace-plot-cpu.c
+++ b/trace-plot-cpu.c
@@ -366,6 +366,7 @@ int cpu_plot_display_info(struct graph_info *ginfo,
 	struct cpu_plot_info *cpu_info = plot->private;
 	struct event_format *event;
 	struct pevent_record *record;
+	struct pevent_record *next_record;
 	struct pevent *pevent;
 	unsigned long sec, usec;
 	const char *comm;
@@ -426,10 +427,18 @@ int cpu_plot_display_info(struct graph_info *ginfo,
 	}
 
 	trace_seq_printf(s, "%lu.%06lu", sec, usec);
-	if (pid)
-		trace_seq_printf(s, " %s-%d", comm, pid);
-	else
-		trace_seq_puts(s, " <idle>");
+
+	next_record = tracecmd_peek_data(ginfo->handle, cpu);
+
+	if (next_record && next_record->missed_events) {
+		trace_seq_puts(s, " MISSED EVENTS");
+	} else {
+
+		if (pid)
+			trace_seq_printf(s, " %s-%d", comm, pid);
+		else
+			trace_seq_puts(s, " <idle>");
+	}
 
 	free_record(record);
 
-- 
2.5.5

